lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -355,7 +355,7 @@ def add_files_to_build_phases(native_target, test_native_targets, app_native_tar
               filter_resource_file_references(file_accessor.resources.flatten) do |compile_phase_refs, resource_phase_refs|
                 native_target.add_file_references(compile_phase_refs, nil)
 
-                if target.build_as_static_framework?
+                if target.build_as_static_framework? && consumer.spec.spec_type == :library
```

Comment:
Nit: can use consumer.spec.library_specification?

Reply:
@dnkoutso you got it!